### PR TITLE
Fix CPU-only import

### PR DESCRIPTION
**Pull request by jloftin-nv**
# Description

import torch_tensorrt currently fails on hosts without an NVIDIA driver because several CUDA queries run at import / def time with no torch.cuda.is_available() guard.

Guard is_tegra_platform(), default_device(), and Device._current_device() so the package imports cleanly on CPU-only machines. GPU paths are unchanged once a driver is present.

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ X] My code follows the style guidelines of this project (You can use the linters)
- [ X] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas and hacks
- [ X] I have made corresponding changes to the documentation
- [X ] I have added tests to verify my fix or my feature
- [ X] New and existing unit tests pass locally with my changes
- [ X] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/_defaults.py [hunk 1]
```diff
@@ -96,4 +96,6 @@
 
 
 def default_device() -> Device:
+    if not torch.cuda.is_available():
+        return Device(gpu_id=0)
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/_defaults.py:101]: why is it `gpu_id=0` instead of `cpu`?

File: py/torch_tensorrt/_Device.py [hunk 2]
```diff
@@ -145,7 +145,7 @@ def _from_torch_device(cls, torch_dev: torch.device) -> Device:
 
     @classmethod
     def _current_device(cls) -> Device:
-        dev_id = torch.cuda.current_device()
+        dev_id = torch.cuda.current_device() if torch.cuda.is_available() else 0
```
Comment (zewenli98) [inline py/torch_tensorrt/_Device.py:148]: same here

File: py/torch_tensorrt/dynamo/_defaults.py (hunk 1 above)
Comment (jloftin-nv) [reply py/torch_tensorrt/dynamo/_defaults.py]: Device actually has no way to represent cpu. So its Either GPU/DLA. Not sure what to choose?

File: py/torch_tensorrt/_Device.py (hunk 2 above)
Comment (jloftin-nv) [reply py/torch_tensorrt/_Device.py]: Device actually has no way to represent cpu. So its Either GPU/DLA. Not sure what to choose?
